aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -753,6 +754,31 @@ func resourceAviatrixTransitGateway() *schema.Resource {
 					},
 				},
 			},
+			"interface_mapping": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of interface names mapped to interface types and indices.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Interface name (e.g., 'eth0', 'eth1').",
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Interface type (e.g., 'wan', 'mgmt').",
+							ValidateFunc: validation.StringInSlice([]string{"wan", "mgmt"}, false),
+						},
+						"index": {
+							Type:        schema.TypeString,
```

Comment:
Is there a reason we do string here and not int?
Side beneifit here is you can do something like:

ValidateFunc: validation.IntAtLeast(0),

Reply:
I tried using a typeInt for index but the interface_mapping started throwing an unexpected value error. When I changed the type to string again, the error went away. My assumption is that the TF provider does not like a complex type map inside a list or set.

Reply:
Updated to int. Added the changes.